art/steps.py

```diff
@@ -235,6 +233,7 @@ def __init__(
 
         self.model_name = model_class.__name__
         self.hash = self.model_class.get_hash()
+        self.trainer = Trainer(**self.trainer_kwargs, logger=self.logger)
```

Comment:
You create trainer twice now  here in __init__ and in __call__

Reply:
Yes, because for now in every call the trainer is initialized once again. I initiatialize it because I need the torch device from it from the init, before the call

Reply:
Okay, doesn't matter, I didn't see the first comment